### PR TITLE
GitHub Action to check if mocks are synced with yea-wandb/master

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
This PR adds a GitHub Action to check if mocks that are used by `wandb/yea-wandb` are in synced with it; to be used as a reminder of sorts to update those, if necessary. 
Also added a note to `CONTRIBUTING.md` with the steps that are currently needed to bring those in sync.

In the meantime, we are thinking about how to make it so we only need to maintain a single copy of those mocks and how to automate the process.

Testing
-------
- Tried screwing up `mock_server.py` (see https://github.com/wandb/yea-wandb/pull/39/commits/0c6d6840550cf14dda7027311304d8c03ecb9aa8).
- Also see [here](https://github.com/wandb/client/runs/4985951300?check_suite_focus=true).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -375,6 +376,22 @@ The circleci uses pytest-split to balance unittest load on multiple nodes. In or
...
+and the [yea-wandb](https://github.com/wandb/yea-wandb) plugin.
+
+The `yea-wandb` plugin for `yea` uses copies of several components from `tests/utils` 
+(`artifact_emu.py`, `mock_requests.py`, and `mock_server.py`) 
+to provide a test environment for functional tests. Currently, we maintain a copy of those components in 
+`yea-wandb/src/yea_wandb`, so they need to be in sync.
+
+If you update one of those files, you need to:
+- Make a new branch in `yea-wandb` with the new versions of the files.
+- Point the client branch you are working on to this `yea-wandb` branch.
```
Comment (kptkin) [inline CONTRIBUTING.md:391]: maybe give more details about where need to point to (I don't think it's clear for people that don't know our workflow)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -375,6 +376,22 @@ The circleci uses pytest-split to balance unittest load on multiple nodes. In or
...
+
+TODO: overview of how to write and run functional tests with [yea](https://github.com/wandb/yea)
+and the [yea-wandb](https://github.com/wandb/yea-wandb) plugin.
+
+The `yea-wandb` plugin for `yea` uses copies of several components from `tests/utils` 
+(`artifact_emu.py`, `mock_requests.py`, and `mock_server.py`) 
+to provide a test environment for functional tests. Currently, we maintain a copy of those components in 
+`yea-wandb/src/yea_wandb`, so they need to be in sync.
+
+If you update one of those files, you need to:
```
Comment (kptkin) [inline CONTRIBUTING.md:389]: maybe add a comment while developing do these 2 steps

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -375,6 +376,22 @@ The circleci uses pytest-split to balance unittest load on multiple nodes. In or
...
+- Make a new branch in `yea-wandb` with the new versions of the files.
+- Point the client branch you are working on to this `yea-wandb` branch.
+- Merge and release `yea-wandb`.
```
Comment (kptkin) [inline CONTRIBUTING.md:392]: Once you are happy with your changes to the above files need to do the following steps.

File: .github/workflows/are-mocks-synced-with-yea-wandb.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: check if mocks are synced with yea-wandb
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  are-mocks-synced:
```
Comment (kptkin) [inline .github/workflows/are-mocks-synced-with-yea-wandb.yml:10]:
```suggestion
  check-mocks-synced:
```
what do you think?

**Comment by dmitryduev** [comment]
Thanks for the review, @kptkin! Mind taking another look?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -375,6 +376,27 @@ The circleci uses pytest-split to balance unittest load on multiple nodes. In or
...
+to provide a test environment for functional tests. Currently, we maintain a copy of those components in 
+`yea-wandb/src/yea_wandb`, so they need to be in sync.
+
+If you update one of those files, you need to:
+- While working on your contribution:
+  - Make a new branch (say, `shiny-new-branch`) in `yea-wandb` and pull in the new versions of the files.
+    Make sure to update the `yea-wandb` version.
+  - Point the client branch you are working on to this `yea-wandb` branch.
+    In `tox.ini`, search for `yea-wandb==<version>` and change it to 
+    `https://github.com/wandb/yea-wandb/archive/shiny-new-branch.zip`.
```
Comment (kptkin) [inline CONTRIBUTING.md:395]: nit: maybe use the git (ssh clone) format?
`git+https://github.com/wandb/yea-wandb.git@...#egg=yea-wandb`

Comment (dmitryduev) [reply CONTRIBUTING.md:395]: Whatever you think is better, I have no preference (and quite frankly don't understand the implications of the difference ;) -- feel free to commit your suggestion :)

**Review by raubitsj — APPROVED** [review]

**Pull request merged**